src/components/augmentedReality/AugmentedReality.js

```diff
@@ -53,16 +53,20 @@ export const AugmentedReality = ({
   useEffect(() => {
     const { settings = {} } = globalSettings;
 
-    try {
-      !!settings.ar &&
-        isARSupportedOnDevice(
-          () => null,
-          () => setIsARSupported(true)
-        );
-    } catch (error) {
-      // if Viro is not integrated, we need to catch the error for `isARSupportedOnDevice of null`
-      console.warn(error);
-    }
+    const supportAr = async () => {
+      try {
+        const isARSupported = (await isARSupportedOnDevice())?.isARSupported;
+
+        if (isARSupported) {
+          setIsARSupported(true);
+        }
+      } catch (error) {
+        // if Viro is not integrated, we need to catch the error for `isARSupportedOnDevice of null`
+        console.error(error);
+      }
+    };
+
+    !!settings.ar?.isARSupported && supportAr();
```

Comment:
there is no `settings.ar?.isARSupported`! you need to remove it. the check with `supportAr()` is enough, right?